homeassistant/components/enasolar/config_flow.py

```diff
@@ -0,0 +1,216 @@
+                }
+            ),
+            errors=_errors,
+            last_step=False,
+        )
+
+    async def async_step_inverter(self, user_input=None) -> FlowResult:
+        """Give the user the opportunities to override inverter config."""
+
+        _errors = {}
+        if user_input is not None:
+            # CAPABILITY is a 9 bit value, bits 1-3 and 9 representing
+            # what features the Innverter has.
+            if user_input[CONF_CAPABILITY] & ~CAPABILITY:
+                _errors[CONF_CAPABILITY] = "capability_invalid"
+            else:
+                self._data.update(user_input)
+                title = self._data[CONF_NAME]
+                return self.async_create_entry(title=title, data=self._data)
+        else:
+            # Use the capability bits from the Inverter. This assumes it
+            # had been possible to actually scrap them from the jScript
+            user_input = {}
+            user_input[CONF_MAX_OUTPUT] = self._enasolar.get_max_output()
+            user_input[CONF_DC_STRINGS] = self._enasolar.get_dc_strings()
+            user_input[CONF_CAPABILITY] = self._enasolar.get_capability()
+
+        return self.async_show_form(
+            step_id="inverter",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_MAX_OUTPUT, default=user_input[CONF_MAX_OUTPUT]
+                    ): vol.In(MAX_OUTPUT),
+                    vol.Required(
+                        CONF_DC_STRINGS, default=user_input[CONF_DC_STRINGS]
+                    ): vol.In(DC_STRINGS),
+                    vol.Required(
+                        CONF_CAPABILITY, default=user_input[CONF_CAPABILITY]
+                    ): vol.Coerce(int),
+                }
+            ),
+            errors=_errors,
+            last_step=True,
+        )
+
+
+class EnaSolarOptionsFlowHandler(config_entries.OptionsFlow):
+    """Allow Polling window to be updated."""
+
+    def __init__(self, config_entry):
+        """Initialize EnaSolar options flow."""
+        self.config_entry = config_entry
+        self._options = dict(config_entry.options)
+
+    async def async_step_init(self, user_input=None) -> FlowResult:
+        """Allow user to reset the Polling times."""
+
+        _errors = {}
+        if user_input is not None:
+            input_valid = True
+            if dt_util.parse_time(user_input[CONF_SUN_UP]) is None:
+                _errors[CONF_SUN_UP] = "time_invalid"
+                input_valid = False
+            if dt_util.parse_time(user_input[CONF_SUN_DOWN]) is None:
+                _errors[CONF_SUN_DOWN] = "time_invalid"
+                input_valid = False
+            if input_valid:
+                if dt_util.parse_time(user_input[CONF_SUN_UP]) >= dt_util.parse_time(user_input[CONF_SUN_DOWN]):  # type: ignore
+                    _errors[CONF_SUN_DOWN] = "time_range"
+                    input_valid = False
```

Comment:
This seems to be used to adjust polling. Why?

Reply:
The inverter's web server crashes if polled to frequently for too long from the same IP address.  My solution was to throttle back the polling frequency of both meters and cumulative data counters separately and to not poll when it was  dark (i.e. no sun, no generation). This is an Enasolar thing not an HA issue.

Reply:
I did consider using the sun integration but for a first try just making them static but configurable seemed simplest .